src/System.Management.Automation/engine/parser/TypeResolver.cs

```diff
@@ -109,6 +117,10 @@ private static Type LookForTypeInAssemblies(TypeName typeName, IEnumerable<Assem
                             }
                         }
                     }
+                    else
+                    {
+                        searchedAssemblies.Add(assembly.FullName);
```

Comment:
If we find that alternate name, then `targetType` from line 91 won't be null. In that case, there is no need to add the current assembly name since the assembly name set is defined to be "A set of assembly names that we have searched from but found no matching type.".

Besides, when `targetType` is not null, we will either return that `targetType` with `exception` being null, or return `null` with a non-null exception in the case of ambiguity from the method. And the search will stop once returned from that method, so `searchedAssemblies` won't be used anymore.